File: vokimi_frontend/src/ts/enums/TestPrivacy.ts
export enum TestPrivacy {
    ForMyself = "for_myself",
    FriendsOnly = "friends_only",
    FriendsAndFollowers = "friends_and_followers",
    Anyone = "anyone"
}

export namespace TestPrivacyUtils {
    export function fromId(id: string): TestPrivacy {
        for (const key in TestPrivacy) {
            if (TestPrivacy[key as keyof typeof TestPrivacy] === id) {
                return TestPrivacy[key as keyof typeof TestPrivacy];
            }
        }
        return TestPrivacy.ForMyself;
    }
    export function getFullName(privacy: TestPrivacy): string {
        switch (privacy) {
            case TestPrivacy.ForMyself:
                return "For Myself";
            case TestPrivacy.FriendsOnly:
                return "Friends Only";
            case TestPrivacy.FriendsAndFollowers:
                return "Friends and Followers";
            case TestPrivacy.Anyone:
                return "Anyone";
            default:
                return "Unknown Privacy Setting";
        }
    }
}

File: vokimi_api/Src/enums/TestPrivacy.cs
namespace vokimi_api.Src.enums
{
    public enum TestPrivacy
    {
        ForMyself,
        FriendsOnly,
        FriendsAndFollowers,
        Anyone
    }
    public static class TestPrivacyExtensions
    {
        public static string GetId(this TestPrivacy privacy) => privacy switch {
            TestPrivacy.ForMyself => "for_myself",
            TestPrivacy.FriendsOnly => "friends_only",
            TestPrivacy.FriendsAndFollowers => "friends_and_followers",
            TestPrivacy.Anyone => "anyone",
            _ => throw new NotImplementedException()
        };
        public static TestPrivacy? FromId(string? id) => id switch {
            "for_myself" => TestPrivacy.ForMyself,
            "friends_only" => TestPrivacy.FriendsOnly,
            "friends_and_followers" => TestPrivacy.FriendsAndFollowers,
            "anyone" => TestPrivacy.Anyone,
            _ => null
        };
    }
}
